crates/bevy_asset/src/io/file_asset_io.rs

```diff
@@ -63,17 +63,17 @@ impl FileAssetIo {
     /// If the `CARGO_MANIFEST_DIR` environment variable is set, then its value will be used
     /// instead. It's set by cargo when running with `cargo run`.
     pub fn get_base_path() -> PathBuf {
-        if let Ok(manifest_dir) = env::var("CARGO_MANIFEST_DIR") {
-            PathBuf::from(manifest_dir)
-        } else {
-            env::current_exe()
-                .map(|path| {
-                    path.parent()
-                        .map(|exe_parent_path| exe_parent_path.to_owned())
-                        .unwrap()
-                })
-                .unwrap()
-        }
+        env::var("CARGO_MANIFEST_DIR")
+            .map(|manifest_dir| PathBuf::from(manifest_dir))
+            .unwrap_or_else(|_| {
+                env::current_exe()
+                    .map(|path| {
+                        path.parent()
+                            .map(|exe_parent_path| exe_parent_path.to_owned())
+                            .unwrap()
+                    })
+                    .unwrap()
+            })
```

Comment:
```suggestion
            .or_else(|| {
                env::current_exe()
                    .and_then(|path| path.parent())
                    .and_then(|parent_path| parent_path.to_owned())
            }).unwrap()
```

Reply:
unwrap_or_else does not ever panic, unless you panic in the closure yourself. Neither will or_else into unwrap, but it will create unreachable code for panicking that may not get optimized out under certain configurations.

Reply:
or_else into unwrap can panic if the closure returns None